modules/_labs/pagination/react/lib/GoTo.tsx

```diff
@@ -0,0 +1,72 @@
+import styled from '@emotion/styled';
+import {type} from '@workday/canvas-kit-react-core';
+import TextInput from '@workday/canvas-kit-react-text-input';
+import React from 'react';
+import uuid from 'uuid/v4';
+
+interface GoToProps {
+  onSubmit: (page: number) => void;
+  max: number;
+  goToLabel?: string;
+}
+const GoToLabel = styled('label')({
+  margin: 'auto 0',
+  paddingRight: '8px',
+  ...type.body,
+  ...type.variant.label,
+});
+
+const GoToWrapper = styled('div')({
+  display: 'flex',
+  paddingLeft: '12px',
+});
+
+const InputWrapper = styled('form')({
+  minWidth: '10px',
+});
+
+const GoTo: React.FC<GoToProps> = props => {
+  const {onSubmit, max, goToLabel} = props;
+  const [value, setValue] = React.useState('');
+
+  const validatePage = (text: string) => {
+    const textAsInteger = parseInt(text, 10);
+    if (textAsInteger < 1) {
+      return 0;
+    }
+    if (textAsInteger > max) {
+      return 0;
+    }
+    return textAsInteger;
+  };
+
+  const goToId = uuid();
```

Comment:
I'm not sure the performance or screen reader implications by this, but every render this id will change. You can use a Ref instead:

https://codesandbox.io/s/p2ndq

```suggestion
  const goToId = React.useRef(uuid()).current;
```